src/intersections/lines.jl

```diff
@@ -43,14 +28,38 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+# compute the parameters of the lines defined by the points a -> b and c -> d 
+# so that the resulting points have minimal distance
+# the ranks help to identify the different types of intersection
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, 
+                             c::Point{N,T}, d::Point{N,T}) where {N,T}
+  # solves the equation (approximately):
+  # a + λ₁ ⋅ v⃗₁ = c + λ₂ ⋅ v⃗₂,
+  # with v⃗₁ = b - a, v⃗₂ = d - c
+  A = [(b - a) (c - d)]
+  y = c - a
+  QR = qr([A y])
 
-  # the intersection point lies in between a and b at a fraction s
-  # (https://en.wikipedia.org/wiki/Line-line_intersection#Formulas)
-  s = (v12[1] * v2[2] - v12[2] * v2[1]) / (v1[1] * v2[2] - v1[2] * v2[1])
+  # calculate the rank of the augmented matrix
+  rₐ = sum(sum(abs.(QR.R), dims = 2) .> atol(T))
+  # calculate the rank of the rectangular matrix
+  r = sum(abs.(diag(QR.R)[1:2]) .> atol(T))
```

Comment:
```suggestion
  r = sum(abs, diag(QR.R)[1:2]) .> atol(T))
```

Same here, can we use `norm` to be more explicit? Performance should be the same right?

Reply:
Here the absolute value of the first diagonal entries is compared with the tolerance. 
The sum counts the `true` values. I don't see how to apply a norm here, except for replacing the sum
over logicals which is not necessary here. I would therefore stick to the suggested code

`r = sum(abs.(diag(QR.R)[1:2]) .> atol(T))`

Example where checking the diagonal (one summation less) does not work, which is the reason for the difference for line 48 and 50:

``` julia
[ -1.0  1.0  0.0
   0.0  0.0  1.0 ]
```

Reply:
The sum of abs is the norm L1, in Julia you can `norm(x, 1)`